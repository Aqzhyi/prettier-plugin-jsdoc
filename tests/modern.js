/* eslint-disable no-undef */
const prettier = require('prettier')

function subject(code, options = {}) {
  return prettier.format(code, {
    parser: 'jsdoc-parser',
    plugins: ['.'],
    jsdocSpaces: 1,
    ...options,
  })
}

test('convert array to modern type', () => {
  const result = subject(`
  /**
   * @typedef {import("react-native-reanimated").default.Adaptable<number>} Adaptable
   * @param {Adaptable} animNode
   * @param {object} InterpolationConfig
   * @param {ReadonlyArray<Adaptable>} InterpolationConfig.inputRange Like [0,1]
   * @param {Array<string>} InterpolationConfig.outputRange Like ["#0000ff","#ff0000"]
   * @param {import("react-native-reanimated").default.Extrapolate} [InterpolationConfig.extrapolate]
   *
   */
  function a(){}
`)

  expect(result).toMatchSnapshot()
})
