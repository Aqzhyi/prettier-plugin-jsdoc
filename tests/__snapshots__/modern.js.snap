// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert array to modern type 1`] = `
"/**
 * @typedef {import(\\"react-native-reanimated\\").default.Adaptable<number>} Adaptable
 * @param {Adaptable} animNode
 * @param {object} InterpolationConfig
 * @param {ReadonlyArray<Adaptable>} InterpolationConfig.inputRange Like [0,1]
 * @param {string[]} InterpolationConfig.outputRange Like
 *     [\\"#0000ff\\",\\"#ff0000\\"]
 * @param {import(\\"react-native-reanimated\\").default.Extrapolate} [InterpolationConfig.extrapolate]
 */
function a() {}
"
`;
